tests/tests/swfs/avm2/throw/Test.as

```diff
@@ -0,0 +1,25 @@
+﻿package {
+	public class Test {
+		public function Test() {}
+	}
+}
+
+import flash.errors.IllegalOperationError;
+
+function foo() {
+	throw new IllegalOperationError("My Error", 30);
+}
+
+try {
+	foo()
+} catch (err) {
+	trace("Caught error!");
+	
+	// FIXME - we currently don't use the original thrown
+	// object in a 'catch' block. Once that's implemented, these
+	// statements will start printing the correct values.
+	trace(err);
+	trace(err.errorID);
```

Comment:
I'm a bit confused why this works in Ruffle currently. Since `err` is currently a string, wouldn't `.errorID` on a `String` object result in an error (instead of `undefined`) in Ruffle?

Reply:
Good catch - this appears to be an issue with how we handle prototype lookups. The prototype of `String` appears to be `Object` (a dynamic class), so calling `get_property_local` will always succeed: https://github.com/ruffle-rs/ruffle/blob/98c0cab6aa1fbbf390c5e3a26bc9865fc4d3beb0/core/src/avm2/object/script_object.rs#L172-L173